test/api/api_test.hpp

```diff
@@ -43,3 +45,15 @@ void print_compare_junction_vectors(std::vector<Junction> const & junctions_expe
                              << junctions_expected_res[i].get_tandem_dup_count() << " == " << junctions_res[i].get_tandem_dup_count() << "\n";
     }
 }
+
+// Helper function for comparing two bamit trees.
+void compare_bamit_trees(std::unique_ptr<bamit::IntervalNode> const & t1, std::unique_ptr<bamit::IntervalNode> const & t2)
+{
+    if (!t1 && !t2) return;
+    if (!t1 || !t2) EXPECT_EQ(1, 0);
```

Comment:
That's true, it's a bit confusing to write it this way. It's basically there so that if a node is NULL in one tree, but contains a valid interval in the other tree, it will fail the test. So I suppose the second option should work!